### PR TITLE
[7.x] AF-2848: Replacement of org.jboss.spec.javax.* deps

### PR DESCRIPTION
**Pull request by mbiarnes**
[JBPM-9742](https://issues.redhat.com/browse/JBPM-9742)

[droolsjbpm-build-bootstrap](https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1626)
[kie-soup](https://github.com/kiegroup/kie-soup/pull/205)
[droolsjbpm-knowledge](https://github.com/kiegroup/droolsjbpm-knowledge/pull/521)
[drools](https://github.com/kiegroup/drools/pull/3628)
[optaplanner](https://github.com/kiegroup/optaplanner/pull/1333)
[appformer](https://github.com/kiegroup/appformer/pull/1143)
[jbpm](https://github.com/kiegroup/jbpm/pull/1917)
[kie-jpmml-integration](https://github.com/kiegroup/kie-jpmml-integration/pull/54)
[jbpm-wb](https://github.com/kiegroup/jbpm-wb/pull/1489)

### Checklist
- [ ] Documentation updated if applicable.
- [ ] Upgrade recipe provided if applicable.

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* for a <b>pull request build</b> please add comment: <b>Jenkins retest this</b>
* for a <b>full downstream build</b> please add comment: <b>Jenkins run fdb</b>
* for a <b>compile downstream build</b> please add comment: <b>Jenkins run cdb</b>
* for a <b>full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>
* for an <b>upstream build</b> please add comment: <b>Jenkins run upstream</b>
* for a <b>Quarkus LTS check</b> please add comment: <b>Jenkins run LTS</b>
* for a <b>Native check</b> please add comment: <b>Jenkins run native</b>
</details>

### CODE REVIEW

**Comment by mbiarnes** [comment]
Jenkins retest this

**Comment by mbiarnes** [comment]
Jenkins retest this

**Comment by mareknovotny** [comment]
@mbiarnes also ask Optaplanner team (@rsynek) for review as they did some of transion to jakarta on master branch too

- https://github.com/kiegroup/optaplanner/pull/834
- https://github.com/kiegroup/optaplanner/pull/836

File: optaplanner-persistence/optaplanner-persistence-jpa/pom.xml [hunk 1]
```diff
@@ -58,14 +58,14 @@
     </dependency>
     <dependency>
       <!-- Replacement for 'org.apache.geronimo.specs:geronimo-jta_1.1_spec' which is excluded from Hibernate deps. -->
-      <groupId>org.jboss.spec.javax.transaction</groupId>
-      <artifactId>jboss-transaction-api_1.2_spec</artifactId>
```
[the remark this answers is not in the document]
Comment (mareknovotny) [reply optaplanner-persistence/optaplanner-persistence-jpa/pom.xml:62]: Are you sure  @rsynek ? this PR is targeting 7.x branch not master

[the remark this answers is not in the document]
Comment (rsynek) [reply optaplanner-persistence/optaplanner-persistence-jpa/pom.xml:62]: Thanks for the nudge, @mareknovotny ; sure I was looking at the master branch. Let me have another look.

[the remark this answers is not in the document]
Comment (mbiarnes) [reply optaplanner-persistence/optaplanner-persistence-jpa/pom.xml:62]: right: on 7.x this doesn't exist

[the remark this answers is not in the document]
Comment (rsynek) [reply optaplanner-persistence/optaplanner-persistence-jpa/pom.xml:62]: Alright, please ignore the part about `DependencyManagement`, which does not make any sense for 7.x branch. The other part still applies, though: there is no import of `javax.transaction.*` except for a single test case.

**Review by ge0ffrey — APPROVED** [review]
LGTM.
Can we do these changes on master/main too if they make sense there?

PR's to 7.x should get the topic prefix [7.x] so it's clear for the reviewers :)

File: optaplanner-persistence/optaplanner-persistence-jpa/pom.xml [hunk 2]
```diff
@@ -58,14 +58,14 @@
...
+      <groupId>jakarta.transaction</groupId>
+      <artifactId>jakarta.transaction-api</artifactId>
       <scope>runtime</scope>
```
[the remark this answers is not in the document]
Comment (mareknovotny) [reply optaplanner-persistence/optaplanner-persistence-jpa/pom.xml:63]: Is my suggestion ok for you @rsynek ?

**Comment by mbiarnes** [comment]
Jenkins restest this

**Comment by mbiarnes** [comment]
Jenkins retest this

**Comment by mbiarnes** [comment]
This PR will be closed since Tiago Bento and William Siqueira are working on the same issue:
https://github.com/kiegroup/optaplanner/pull/1463

**Pull request closed without merging**